doc/engine/Architecture.md

```diff
@@ -0,0 +1,126 @@
+Architecture of the OpenQuake Engine v2
+=========================================
+
+The OpenQuake Engine version 2.0 is a complete rewrite of version
+1.0, so a new document describing the overall architecture is
+needed. Whereas in the past the Engine was database-centric and
+structured as a Web application with an Object Relational Mapper, now
+it is calculation-centric and structured as a scientific application:
+everything is done in memory and in the core
+engine there is no database, nor ORM. Intermediate results are stored
+in HDF5 format. In the calculators we work as much as possible in
+terms of arrays which are efficiently manipulated at C/Fortran speed
+with a stack of well established scientific libraries (numpy/scipy).
+
+Design principles
+-----------------
+
+The main design principle has been *simplicity*: everything has to be
+as simple as possible (but not simplest). The goal has been to keep
+the engine simple enough that a single person can understand it, can
+debug it and can extend it without tremendous efforts. All the rest
+come from simplicity: transparency, ability to inspect and debug, modularity,
+adaptability of the code, etc. Even efficiency: in the last three
+years most of the performance improvements came from free, just from
+removing complications. When a thing is simple it is easy to make it
+fast. The battle for simplicity is never ending, so there are still
+several things in the engine that are more complex that they should:
+we are working on that.
+
+After simplicity the second design goal has been *performance*: the
+engine is a number chrunching application after all, and we need to run
+massively parallel calculations taking days or weeks of
+runtime. Efficiency in terms of computation time and memory
```

Comment:
You mentioned efficiency also while writing about simplicity. Perhaps you could move that part here.